### PR TITLE
feat(scripts): add additional webpack plugins

### PR DESCRIPTION
**Pull request by wincent**
As requested in https://github.com/liferay/liferay-frontend-guidelines/issues/156.

Added:

- [html-webpack-plugin](https://www.npmjs.com/package/html-webpack-plugin) to create/modify html files.
- [mini-css-extract-plugin](https://www.npmjs.com/package/mini-css-extract-plugin) to extract CSS.
- [script-ext-html-webpack-plugin](https://www.npmjs.com/package/script-ext-html-webpack-plugin) to inline scripts.
- [style-ext-html-webpack-plugin](https://www.npmjs.com/package/style-ext-html-webpack-plugin) to inline CSS.
  
We [gathered some basic data](https://github.com/liferay/liferay-frontend-guidelines/issues/156#issuecomment-628153969) about the proposed plugins to evaluate their risk (from least to most sketchy):

- https://github.com/webpack-contrib/mini-css-extract-plugin
  - 7.8m weekly downloads
  - 3.9k GitHub stars
  - 59 open issues
  - 26 open (non-bot) PRs
  - Last commit: Dec 2019
  - Last release: Dec 2019
  - 4 top-level items in dependency graph.
- https://github.com/jantimon/html-webpack-plugin
  - 2.2m weekly downloads
  - 8.8k GitHub stars
  - 33 open issues
  - 6 open (non-bot) PRs
  - Last commit: May 2020
  - Last release: Apr 2020
  - 9 top-level items in dependency graph.
- https://github.com/numical/script-ext-html-webpack-plugin
  - 123k weekly downloads
  - 497 GitHub stars
  - 15 open issues
  - 1 open (non-bot) PR
  - Last commit: July 2019
  - Last tagged release: Nov 2018; last non-tagged release: July 2019 ([issue](https://github.com/numical/script-ext-html-webpack-plugin/issues/76))
  - 1 top-level item in dependency graph.
- https://github.com/numical/style-ext-html-webpack-plugin
  - 33k weekly downloads
  - 120 GitHub stars
  - 6 open issues
  - 0 open (non-bot) PRs
  - Last commit: July 2019
  - Last tagged release: Jan 2018; last non-tagged release: July 2019 ([issue](https://github.com/numical/style-ext-html-webpack-plugin/issues/51)) 
  - 2 top-level items in dependency graph.

No glaring red flags there, and only really a yellow flag to note the lack of tags for the numical plugins (and the non-response to the issues that I opened about this over the last 20 days). All in all though, we left the request open for a few weeks to give people a chance to comment on it and nobody did, and these dependencies are relatively self contained (in the sense that they are webpack-only, and internal-only, so we can change our mind about them in the future), so we're going to merge them in.

Done with:

    yarn add \
      html-webpack-plugin@4.3.0 \
      script-ext-html-webpack-plugin@2.1.4 \
      mini-css-extract-plugin@0.9.0 \
      style-ext-html-webpack-plugin@4.1.2

Note that the original request mentioned v2.1.1 of the script-ext-html-webpack-plugin, but looking at the changes from there up to v2.1.4, I believe the newer version should be compatible.

`yarn audit` before:

    3391 vulnerabilities found - Packages audited: 2469
    Severity: 3340 Low | 38 Moderate | 13 High

`yarn audit` after:

    3393 vulnerabilities found - Packages audited: 2486
    Severity: 3342 Low | 38 Moderate | 13 High

`yarn audit` diff:

```
diff --git a/tmp/before b/tmp/after
index 557a697..2ca1bc6 100644
--- a/tmp/before
+++ b/tmp/after
@@ -38704,6 +38704,20 @@ yarn audit v1.22.4
 ├───────────────┼──────────────────────────────────────────────────────────────┤
 │ Dependency of │ liferay-npm-scripts                                          │
 ├───────────────┼──────────────────────────────────────────────────────────────┤
+│ Path          │ liferay-npm-scripts > html-webpack-plugin > loader-utils >   │
+│               │ json5 > minimist                                             │
+├───────────────┼──────────────────────────────────────────────────────────────┤
+│ More info     │ https://www.npmjs.com/advisories/1179                        │
+└───────────────┴──────────────────────────────────────────────────────────────┘
+┌───────────────┬──────────────────────────────────────────────────────────────┐
+│ low           │ Prototype Pollution                                          │
+├───────────────┼──────────────────────────────────────────────────────────────┤
+│ Package       │ minimist                                                     │
+├───────────────┼──────────────────────────────────────────────────────────────┤
+│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
+├───────────────┼──────────────────────────────────────────────────────────────┤
+│ Dependency of │ liferay-npm-scripts                                          │
+├───────────────┼──────────────────────────────────────────────────────────────┤
 │ Path          │ liferay-npm-scripts > liferay-lang-key-dev-loader >          │
 │               │ loader-utils > json5 > minimist                              │
 ├───────────────┼──────────────────────────────────────────────────────────────┤
@@ -38718,6 +38732,20 @@ yarn audit v1.22.4
 ├───────────────┼──────────────────────────────────────────────────────────────┤
 │ Dependency of │ liferay-npm-scripts                                          │
 ├───────────────┼──────────────────────────────────────────────────────────────┤
+│ Path          │ liferay-npm-scripts > mini-css-extract-plugin > loader-utils │
+│               │ > json5 > minimist                                           │
+├───────────────┼──────────────────────────────────────────────────────────────┤
+│ More info     │ https://www.npmjs.com/advisories/1179                        │
+└───────────────┴──────────────────────────────────────────────────────────────┘
+┌───────────────┬──────────────────────────────────────────────────────────────┐
+│ low           │ Prototype Pollution                                          │
+├───────────────┼──────────────────────────────────────────────────────────────┤
+│ Package       │ minimist                                                     │
+├───────────────┼──────────────────────────────────────────────────────────────┤
+│ Patched in    │ >=0.2.1 <1.0.0 || >=1.2.3                                    │
+├───────────────┼──────────────────────────────────────────────────────────────┤
+│ Dependency of │ liferay-npm-scripts                                          │
+├───────────────┼──────────────────────────────────────────────────────────────┤
 │ Path          │ liferay-npm-scripts > sass-loader > loader-utils > json5 >   │
 │               │ minimist                                                     │
 ├───────────────┼──────────────────────────────────────────────────────────────┤
@@ -53988,6 +54016,6 @@ yarn audit v1.22.4
 ├───────────────┼──────────────────────────────────────────────────────────────┤
 │ More info     │ https://www.npmjs.com/advisories/1500                        │
 └───────────────┴──────────────────────────────────────────────────────────────┘
-3391 vulnerabilities found - Packages audited: 2469
-Severity: 3340 Low | 38 Moderate | 13 High
-Done in 5.20s.
+3393 vulnerabilities found - Packages audited: 2486
+Severity: 3342 Low | 38 Moderate | 13 High
+Done in 4.46s.
```

In short, I don't seen anything too concerning there.

### CODE REVIEW

**Comment by wincent** [comment]
For some reason GitHub isn't showing this as green, but it is:

Screenshot 2020-06-02 -131221-iBV25WkM@2x

So I'm going to merge it.
